dispatch/static/manager/src/js/actions/GenericActions.js

```diff
@@ -0,0 +1,132 @@
+import { normalize, arrayOf } from 'normalizr'
+import { push } from 'react-router-redux'
+
+export default class GenericActions {
+
+  constructor(types, api, schema) {
+    this.types = types
+    this.api = api
+    this.schema = schema
+  }
+
+  prepareData(data) {
+    return data
+  }
+
+  get(token, id, params) {
+    return {
+      type: this.types.GET,
+      payload: this.api.get(token, id, params)
+        .then(json => normalize(json, this.schema))
+    }
+  }
+
+  list(token, params) {
+    return {
+      type: this.types.LIST,
+      payload: this.api.list(token, params)
+        .then(json => ({
+          count: json.count,
+          results: normalize(json.results, arrayOf(this.schema))
+        }))
+    }
+  }
+
+  save(token, id, data) {
+    return {
+      type: this.types.SAVE,
+      payload: this.api.save(token, id, this.prepareData(data))
+        .then(json => normalize(json, this.schema))
+    }
+  }
+
+  create(token, id, data) {
+    return {
+      type: this.types.CREATE,
+      payload: this.api.create(token, this.dataAdapter(data))
+        .then(json => normalize(json, this.schema))
+    }
+  }
+
+  delete(token, id, next=null) {
+    return (dispatch) => {
```

Comment:
another nit, but half of the single-var arrow fns seem to have `(arg) =>` and the other half don't (`arg =>`). is there a pattern there?